xrpl/models/requests/amm_info.py

```diff
@@ -0,0 +1,36 @@
+"""This request retrieves information about an AMM instance."""
+from __future__ import annotations
+
+from dataclasses import dataclass, field
+from typing import Dict
+
+from xrpl.models.currencies import Currency
+from xrpl.models.requests.request import Request, RequestMethod
+from xrpl.models.required import REQUIRED
+from xrpl.models.utils import require_kwargs_on_init
+
+
+@require_kwargs_on_init
+@dataclass(frozen=True)
+class AMMInfo(Request):
+    """
+    This request retrieves information about an AMM instance.
+
+    Must provide Asset and Asset2 params.
+    """
+
+    asset: Currency = REQUIRED  # type: ignore
+    """
+    Specifies one of the pool assets (XRP or token) of the AMM instance.
+    """
+
+    asset2: Currency = REQUIRED  # type: ignore
+    """
+    Specifies the other pool asset of the AMM instance.
+    """
+
+    method: RequestMethod = field(default=RequestMethod.AMM_INFO, init=False)
+
+    def _get_errors(self: AMMInfo) -> Dict[str, str]:
```

Comment:
This is unneeded - request models aren't checked and also this is just calling the `super._get_errors` function anyways, which happens by default.